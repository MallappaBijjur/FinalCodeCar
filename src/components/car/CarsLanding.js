import React from 'react';
import '../home-car.css';
import CarItem from './CarItem';
import Pagination from "../common/Pagination";

const CarsLanding = props => {
	const { cars, totalCarsCount } = props;

	const availableInfo = `Showing ${totalCarsCount > 10 ? 10: totalCarsCount} of ${totalCarsCount} results`

	return(<div className="landing">
			<div className="landing__top">
				<p className="landing__top--text--bold" id="available cars">Available Cars</p>
				<p className="landing__top--info">{availableInfo}</p>
			</div>
			<div data-testid="cars-items">
				{
					cars && cars.map(item => <CarItem key={item.stockNumber} {...item} />)
				}
			</div>
			<Pagination {...props}/>
		</div>);
};

export default CarsLanding;