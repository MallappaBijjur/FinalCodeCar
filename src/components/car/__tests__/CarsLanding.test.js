import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import rerender from 'react-test-renderer';
import CarsLanding from '../CarsLanding';

describe("CarsLanding", () => {

	const originalError = console.error
	beforeAll(() => {
		console.error = (...args) => {
			if (/Warning.*not wrapped in act/.test(args[0])) {
				return
			}
			originalError.call(console, ...args)
		}
	})

	afterAll(() => {
		console.error = originalError
	});

	afterEach(cleanup);
	
    const carsData = [
        {
            "stockNumber":30562,
            "manufacturerName":"Fiat",
            "modelName":"Ducato",
            "color":"yellow",
            "mileage":{
                    "number":100013,
                    "unit":"km"
            },
            "fuelType":"Petrol",
            "pictureUrl":"https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
		},
        {
            "stockNumber":35592,
            "manufacturerName":"Chrysler",
            "modelName":"New Yorker",
            "color":"white",
            "mileage":{
                    "number":100115,
                    "unit":"km"
            },
            "fuelType":"Petrol",
            "pictureUrl":"https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        }];
    
	it("renders without crashing", ()=>{
		const div = document.createElement("div");
		ReactDOM.render(<CarsLanding  cars={carsData} totalCarsCount={2} />, div);
	});
	
	it('renders CarsLanding', () => {
        const { queryByTestId } = render(<CarsLanding cars={carsData} totalCarsCount={2} />);   
		const linkElement = queryByTestId(/cars-items/i);
		expect(linkElement).toBeInTheDocument();
	});
	
	it("matches snapshot", ()=> {
		const tree = rerender.create(<CarsLanding cars={carsData} totalCarsCount={2} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
