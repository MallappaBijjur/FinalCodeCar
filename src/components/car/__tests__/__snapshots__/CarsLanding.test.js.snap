// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarsLanding matches snapshot 1`] = `
<div
  className="landing"
>
  <div
    className="landing__top"
  >
    <p
      className="landing__top--text--bold"
      id="available cars"
    >
      Available Cars
    </p>
    <p
      className="landing__top--info"
    >
      Showing 2 of 2 results
    </p>
  </div>
  <div
    data-testid="cars-items"
  >
    <div
      className="car__item"
      data-testid="car-30562"
    >
      <div
        className="car__item__img--container"
      >
        <img
          alt="Car"
          className="car__item--img"
          src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        />
      </div>
      <div
        className="car__item__details--container"
      >
        <div
          className="car__item__details"
        >
          <p
            className="car__item--info landing__top--text--bold margin__set"
          >
            Ducato
          </p>
          <div
            className="car__item__text car__item__text--small car__item__width--max "
          >
            <p
              className="car__item__text"
            >
              Stock #
              30562 -
              100013 KM -
              Petrol -
              yellow
            </p>
          </div>
          <a
            className="car__item__text car__item__text--link car__item__text--small pagelink__text--orange"
            href="/car/30562"
          >
            View details
          </a>
        </div>
      </div>
    </div>
    <div
      className="car__item"
      data-testid="car-35592"
    >
      <div
        className="car__item__img--container"
      >
        <img
          alt="Car"
          className="car__item--img"
          src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
        />
      </div>
      <div
        className="car__item__details--container"
      >
        <div
          className="car__item__details"
        >
          <p
            className="car__item--info landing__top--text--bold margin__set"
          >
            New Yorker
          </p>
          <div
            className="car__item__text car__item__text--small car__item__width--max "
          >
            <p
              className="car__item__text"
            >
              Stock #
              35592 -
              100115 KM -
              Petrol -
              white
            </p>
          </div>
          <a
            className="car__item__text car__item__text--link car__item__text--small pagelink__text--orange"
            href="/car/35592"
          >
            View details
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="pagination"
  >
    <a
      className="link__item--text link__item--text--small pagelink__text--orange"
      href="#available cars"
      onClick={[Function]}
    >
      First
    </a>
    <a
      className="link__item--text link__item--text--small pagelink__text--orange"
      href="#available cars"
      onClick={[Function]}
    >
      Previous
    </a>
    <a
      className="link__item--text link__item--text--small link__text--black"
      href="#available cars"
    >
      Page undefined of undefined
    </a>
    <a
      className="link__item--text link__item--text--small pagelink__text--orange"
      href="#available cars"
      onClick={[Function]}
    >
      Next
    </a>
    <a
      className="link__item--text link__item--text--small pagelink__text--orange"
      href="#available cars"
      onClick={[Function]}
    >
      Last
    </a>
  </div>
</div>
`;
